plugins/common-aliases/common-aliases.plugin.zsh

```diff
@@ -54,27 +54,110 @@ autoload -Uz is-at-least
 if is-at-least 4.2.0; then
   # open browser on urls
   if [[ -n "$BROWSER" ]]; then
-    _browser_fts=(htm html de org net com at cx nl se dk)
+    _browser_fts=(htm html xhtml)
     for ft in $_browser_fts; do alias -s $ft='$BROWSER'; done
   fi
 
-  _editor_fts=(cpp cxx cc c hh h inl asc txt TXT tex)
-  for ft in $_editor_fts; do alias -s $ft='$EDITOR'; done
+  # open editable text files in text editor
+  if [[ -n "$VISUAL" ]] || [[ -n "$EDITOR" ]]; then
+    _aliases_launch_editor() {
+      # directly launch matched files with executable bit and shebang present
+      if [[ -x "${1}" ]]; then
+        read -r <"${1}"
+        if [[ ${#REPLY} -ge 3 ]] && [[ ${REPLY:0:2} = '#!' ]]; then
+          "$@"
+          return $?
+        fi
+      fi
```

Comment:
I think we can ignore whether the file has a shebang and execute it directly if it has an execute-bit, which is the behavior we'd get if no suffix-alias was present.